stack/stack_structs.h

```diff
@@ -0,0 +1,26 @@
+#ifndef STACK_STRUCTS_H
+#define STACK_STRUCTS_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <float.h>
+#include <math.h>
+#include <assert.h>
+#include <stdint.h>
+
+#define CANARY_VALUE 0xDEDA
```

Comment:
don't use `#define` when you can use constants. you can use `#define` only when it's really necessary. also, don't use macros when you can use function. every usage of `#define` should be VERY reasonable

Reply:
resolved, but stayed define because of ded did like this